src/v/kafka/server/handlers/create_partitions.cc

```diff
@@ -180,6 +180,18 @@ ss::future<response_ptr> create_partitions_handler::handle(
           return ctx.authorized(security::acl_operation::alter, tp.name);
       });
 
+    if (!ctx.audit()) {
+        auto distance = std::distance(
+          request.data.topics.begin(), valid_range_end);
+
+        co_return co_await ctx.respond(create_partitions_response(
+          error_code::broker_not_available,
+          "Broker not available - audit system failure",
+          std::move(resp),
+          std::move(request),
+          distance));
```

Comment:
Would `valid_range_end` still be a valid iterator after moving the structure (`request`) that the `valid_range_end` iterator was originally pointing to?

Reply:
i mean the type would also need to change to something other than difference_type. no need to change, just pointing out that this can be done the same way as in the previous audit PR that avoids distance().